include/buffer.h

```diff
@@ -62,35 +60,62 @@ class buffer {
 	~buffer() {}
 
 	template <access_mode Mode, typename Functor, int D = Dims, std::enable_if_t<(D > 0), int> = 0>
-	accessor<DataT, Dims, Mode, target::device> get_access(handler& cgh, Functor rmfn) const {
+	accessor<DataT, Dims, Mode, target::device> get_access(handler& cgh, Functor rmfn) {
+		return get_access<Mode, target::device, Functor>(cgh, rmfn);
+	}
+
+	template <access_mode Mode, typename Functor, int D = Dims, std::enable_if_t<D == 0, int> = 0>
+	accessor<DataT, Dims, Mode, target::device> get_access(handler& cgh) {
+		return get_access<Mode, target::device, Functor>(cgh);
+	}
+
+	template <access_mode Mode, target Target, typename Functor, int D = Dims, std::enable_if_t<(D > 0), int> = 0>
+	accessor<DataT, Dims, Mode, Target> get_access(handler& cgh, Functor rmfn) {
+		return accessor<DataT, Dims, Mode, Target>(*this, cgh, rmfn);
+	}
+
+	template <access_mode Mode, target Target, typename Functor, int D = Dims, std::enable_if_t<D == 0, int> = 0>
+	accessor<DataT, Dims, Mode, Target> get_access(handler& cgh) {
+		return accessor<DataT, Dims, Mode, Target>(*this, cgh);
+	}
+
+	template <access_mode Mode, typename Functor, int D = Dims, std::enable_if_t<(D > 0), int> = 0>
+	[[deprecated("Calling get_access on a const buffer is deprecated")]] accessor<DataT, Dims, Mode, target::device> get_access(
```

Comment:
Since after Fabian's remark below there's only two overloads left, which are also right next to each other, I feel like adding the macro would make things more complicated than necessary.